scripts/floods/raster_to_csv.py

```diff
@@ -0,0 +1,1149 @@
+            # Check if data point is allowed by the ignore/allow masks.
+            if ignore_arr is not None:
+                ignore_point = get_raster_data_latlng(ignore_r, ignore_arr,
+                                                      data['latitude'],
+                                                      data['longitude'])
+                if ignore_point:
+                    logging.debug(
+                        f'Ignoring {data} due to ignore mask: {ignore_point}')
+                    _add_counter(counter, 'ignored_points', 1)
+                    continue
+            if allow_arr is not None:
+                allow_point = get_raster_data_latlng(allow_r, allow_arr,
+                                                     data['latitude'],
+                                                     data['longitude'])
+                if not allow_point:
+                    logging.debug(
+                        f'Ignoring {data} due to allow mask: {allow_point}')
+                    _add_counter(counter, 'not_allowed_points', 1)
+                    continue
+            if add_data_point(data_points, data, config, data_filter, counter):
+                _add_counter(counter, 'output_data_points', 1)
+            _show_counters(counter, config)
+    _show_counters(counter)
+    if output_csv:
+        write_data_csv(data_points, output_csv, [], config, counter)
+    return True
+
+
+def _estimate_num_rows(filename: str) -> int:
+    '''Returns the estimated number of rows based on size of the first few rows.'''
+    filesize = os.path.getsize(filename)
+    with open(filename) as fp:
+        lines = fp.read(4000)
+    line_size = len(lines) / (lines.count('\n') + 1)
+    return filesize / line_size
+
+
+def process_csv_points(input_csv: str,
+                       ignore_csv: str = None,
+                       allow_csv: str = None,
+                       data_points: dict = None,
+                       config: dict = {},
+                       output_csv: str = None,
+                       counter: dict = None) -> bool:
+    '''Process data points from CSV.
+    Args:
+      input_csv: Input csv file to load data points.
+         Assumed to have columns: latitude, longitude
+      ignore_csv: CSV file with points to be ignored.
+      allow_csv: CSV file with points to be allowed.
+      data_points: Dictionary of data points keyed by location+date.
+        into which new points from input_csv are added.
+      config: dictionary of configuration parameter:values.
+      output_csv: Output csv file if data is to be saved.
+      counter: dictionary of named counter:values.
+    Returns:
+      True if data was processed.
+   '''
+    if data_points is None:
+        data_points = {}
+    if counter is None:
+        counter = {}
+    ignore_points = {}
+    if ignore_csv:
+        filter_config = dict(config)
+        filter_config['ignore_date'] = True
+        ignore_points = process_csv_points(input_csv=ignore_csv,
```

Comment:
I slightly find the pattern of recursive calls with different configs configs nifty, but hard to follow.  A lot of what's going on is opaque in the configs :(